private/bufpkg/bufconfig/buf_yaml_file.go

```diff
@@ -697,14 +759,23 @@ func writeBufYAMLFile(writer io.Writer, bufYAMLFile BufYAMLFile) error {
 		for _, moduleConfig := range bufYAMLFile.ModuleConfigs() {
 			moduleDirPath := moduleConfig.DirPath()
 			joinDirPath := func(importPath string) string {
-				return filepath.Join(moduleDirPath, importPath)
+				return normalpath.Join(moduleDirPath, importPath)
```

Comment:
This is a fix, because other paths, such as those in lint/breaking configs are handled using `normalpath.Join`.